### ISSUE TITLE
Offline feature unable to download tiles for custom Styles

### ISSUE THREAD

**Issue by Guimareshh** [mapbox/mapbox-maps-android#1789]
## Environment

- Android OS version: all
- Devices affected: all
- Maps SDK Version: 10.9.0

## Observed behavior and steps to reproduce
Moving from legacy SDK (9.7.2) to new SDK (10.9.0) I reworked the Offline feature of my app to use the new APIs as stated on the [official documentation](https://docs.mapbox.com/android/maps/guides/offline/). Offline is working fine for Outdoor and Satellites styles, but for custom styles (we are using IGN and SwissTopo for instance) the SDK behaves as it's working but no tiles have been downloaded.

Indeed all properties from TileRegionLoadProgress are set to 0 and TileRegionCallback.run is set with `region.error` null and  TileRegion not null as seen on this screenshot:

<img width="898" alt="Screenshot 2022-11-01 at 14 21 31" src="https://user-images.githubusercontent.com/9654353/199244192-3c500fc1-220f-4a05-a52c-f30049c8548b.png">

## Expected behavior
Tiles from custom styles should be properly downloaded.

**Comment by ZiZasaurus** [comment]
@Guimareshh thank you for reaching out. When you reference custom styles, are these styles you are loading locally?

**Comment by Guimareshh** [comment]
@ZiZasaurus Custom styles are loaded from an URL (`https://example.com/custom-style.json`)

**Comment by ZiZasaurus** [comment]
Ah, understood. Do your those custom styles rely on Mapbox tilesets?

**Comment by Guimareshh** [comment]
Those custom styles don't rely on Mapbox tilesets. Is that a issue with this SDK?

**Comment by ZiZasaurus** [comment]
@Guimareshh, ah yes, thank you for clarifying. The tile pack API can only work with Mapbox tilesets. https://docs.mapbox.com/android/maps/guides/offline/#supported-tiled-sources

**Comment by Guimareshh** [comment]
Ah thanks for pointing to the documentation, I didn't saw it. 

Unfortunately, it's a huge drawback for us, as the offline feature with custom styles is a huge part of our app. Is this being work out and will be supported any time soon?

**Comment by Guimareshh** [comment]
Our use case is the following: we allow our users to view a hike path on a mapview. They can switch between styles including custom ones not using Mapbox tilesets. And they can download these information offline (hike path, map styles ...). In some European countries it's a common use case to use map styles from national mapping agencies (for instance in France we have IGN, in Switzerland SwissTopo etc...). 

Can we safely use the deprecated OfflineManager in the SDK v10 until this feature is supported with the new APIs?

**Comment by ZiZasaurus** [comment]
@Guimareshh, yes the legacy tile-based offline system, which remains available in v10, will allow you to accomplish this use case.

**Comment by Guimareshh** [comment]
Is there an official issue following a support for custom styles being download with the new offline APIs? Otherwise this issue shouldn't be closed imo

**Comment by VahanLab** [comment]
I'm also interested in this use case @ZiZasaurus.
@Guimareshh are you really able to download an IGN raster layer using the deprecated API? As far I can see, the OfflineRegionGeometryDefinition class used in the OfflineRegionManager also uses a styleURI, so I guess it can only work with mapbox custom styles as well

**Comment by davenquinn** [comment]
Similar to many organizations, [Macrostrat](https://macrostrat.org) maintains its own tile services. We'd like to use Mapbox's caching system for these tiles, and we hope to avoid supporting both the "legacy" and modern caching systems (currently, we stick entirely with the legacy system).

Is there a way that we can move in this direction by exposing a "tile packs" API in our system (presuming we could replicate the packaging format)? Or is tile pack provision explicitly limited to Mapbox's own servers?

**Comment by davenquinn** [comment]
@ZiZasaurus or others at Mapbox, some guidance on this would be appreciated. We are considering implementing our own parallel caching approach for several geology-focused apps (Rockd, StraboSpot, and Mapboard GIS), because of the limitations of Mapbox's system for our hosted data layers. Unfortunately, the _Host your tiles with Mapbox_ solution is not aligned with our business goals and application design.

Given the engineering involved, it would be fantastic if we would be able to continue working with Mapbox's tile caching system instead. This would also keep us more firmly tied to Mapbox's ecosystem/tools than charting our own path.

**Comment by timbtimbtimb** [comment]
I am joining everyone in saying that this is a huge drawback, and I'll add that it feels to me like a shady business practice, forcing people to host tiles on your service (which might not even be legal for us to do in the case of French IGN tiles mentioned by others).
Please consider leaving open custom styles for offline download.
